costmodel/router.go

```diff
@@ -260,169 +326,100 @@ func (a *Accesses) AggregateCostModel(w http.ResponseWriter, r *http.Request, ps
-		// in which case it shifts endTime back by given duration
-		endTime := time.Now()
-		if offset != "" {
-			o, err := time.ParseDuration(offset)
-			if err != nil {
-				klog.V(1).Infof("error parsing offset: %s", err)
-				w.Write(wrapData(nil, err))
-				return
-			}
-			endTime = endTime.Add(-1 * o)
-		}
-
-		if a.ThanosClient != nil {
-			if endTime.After(time.Now().Add(-3 * time.Hour)) {
-				klog.Infof("Setting end time backwards to first present data")
-				endTime = time.Now().Add(-3 * time.Hour)
-			}
-		}
-
-		// if window is defined in terms of days, convert to hours
-		// e.g. convert "2d" to "48h"
-		window, err := normalizeTimeParam(window)
-		if err != nil {
-			w.Write(wrapData(nil, err))
-			return
-		}
-
-		// convert time window into start and end times, formatted
-		// as ISO datetime strings
-		dur, err = time.ParseDuration(window)
-		if err != nil {
-			w.Write(wrapData(nil, err))
-			return
-		}
-		startTime := endTime.Add(-1 * dur)
-
-		start = startTime.Format(layout)
-		end = endTime.Format(layout)
-
-		klog.V(1).Infof("start: %s, end: %s", start, end)
+	// field by which to aggregate must be set, with subfield(s) required when the field
+	// is "label" to define the specific labels by which to aggregate.
+	subfields := []string{}
+	if len(subfieldStr) > 0 {
+		subfields = strings.Split(r.URL.Query().Get("aggregationSubfield"), ",")
 	}
-
-	// dataCount is the number of time series data expected for the given interval,
-	// which we compute because Prometheus time series vectors omit zero values.
-	// This assumes hourly data, incremented by one to capture the 0th data point.
-	dataCount := int64(dur.Hours()) + 1
-	klog.V(1).Infof("for duration %s dataCount = %d", dur.String(), dataCount)
-
-	// aggregation field is required
 	if field == "" {
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write(wrapData(nil, fmt.Errorf("Missing aggregation field parameter")))
 		return
 	}
-
-	// aggregation subfield is required when aggregation field is "label"
 	if field == "label" && len(subfields) == 0 {
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write(wrapData(nil, fmt.Errorf("Missing aggregation subfield parameter for aggregation by label")))
 		return
 	}
 
-	// enforce one of four available rate options
+	// if rate is defined, ensure it is a valid option
 	if rate != "" && rate != "hourly" && rate != "daily" && rate != "monthly" {
 		w.WriteHeader(http.StatusBadRequest)
 		w.Write(wrapData(nil, fmt.Errorf("If set, rate parameter must be one of: 'hourly', 'daily', 'monthly'")))
 		return
 	}
 
+	// enable remote if it is available and not disabled
+	remoteAvailable := os.Getenv(remoteEnabled) == "true"
+	remoteEnabled := remote && remoteAvailable
+	klog.Infof("REMOTE ENABLED: %t", remoteEnabled)
+
+	// default to given Prometheus client, but use Thanos Client if it
+	// exists and remote is not disabled
+	pClient := a.PrometheusClient
+	if remoteEnabled && a.ThanosClient != nil {
```

Comment:
just use a.ThanosClient != nil here. The remoteEnabled flag is for postgres. ThanosClient will only be instantiated if Thanos is available.

Reply:
That would be a change from master (see https://github.com/kubecost/cost-model/blob/master/costmodel/router.go#L414). Is that still the intended logic?